xla/hlo/ir/hlo_module.h

```diff
@@ -711,6 +718,11 @@ class HloModule {
       std::unique_ptr<HloComputation> computation, bool is_entry,
       bool uniquify_identifiers, bool preserve_entry_layouts);
 
+  // Performs a deep clone of current module to context->module, and populate
```

Comment:
I think this can be a private method. I guess the new method CloneWithContext serves the need to get access to the mappings?

Reply:
Yes, it is a private method now. The public method will only have the other two:

```
std::unique_ptr<HloModule> Clone(
      const std::string& suffix = "clone",
      std::optional<const HloModuleConfig> config = std::nullopt) const;

 
  std::pair<std::unique_ptr<HloModule>, std::unique_ptr<HloCloneContext>>
  CloneWithContext(
      const std::string& suffix,
      std::optional<const HloModuleConfig> config = std::nullopt) const;
```

You mean this is OK?